### PR TITLE
[Plugin] opencvmini api expose resize function, update normalize

### PR DESCRIPTION
**Pull request by Charles-Schleich**
Added resize function and updated normalize to more closely match the opencv calling standard,
I have not matched it 100% as for my usecase, i dont see a time where i will need to pass in a `mask`, and use the `dtype` flag
https://docs.opencv.org/3.4/d2/de8/group__core__array.html#ga87eef7ee3970f86906d69a92cbf064bd
However if the plugin should be designed to match opencv exactly in how it is used, then i am happy to change this.

And maybe this is a good place to ask whether or not we should maybe expose a way to query the dimensions for certain `MatKey` from the runtime ?
It might be useful for whoever is calling these functions, should they manipulate dimensions internally, and we might want to put them into a buffer from whichever language is calling them.

Otherwise, let me know if there are any changes that I can / should make.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by hydai** [comment]
Hi @dannypsnl 
Could you please help review this?

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp [hunk 1]
```diff
@@ -123,14 +123,16 @@ Expect<void> WasmEdgeOpenCVMiniImencode::body(
...
   auto Src = Env.getMat(SrcMatKey);
   if (!Src) {
     return Unexpect(ErrCode::Value::HostFuncError);
   }
   cv::Mat Dst;
-  // convert each elements `v` of `Src` to `(1/255) * v + 0`
-  Src->convertTo(Dst, CV_32F, 1. / 255., 0.);
+  // -1 below is the dtype: We want the output array to have same type as input
+  // array
+  cv::normalize(*Src, Dst, Alpha, Beta, NormType, -1, cv::noArray());
```
Comment (dannypsnl) [inline plugins/wasmedge_opencvmini/opencvmini_func.cpp:135]: hard coded arguments `-1` and `cv::noArray()`

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp [hunk 2]
```diff
@@ -180,5 +182,21 @@ Expect<uint32_t> WasmEdgeOpenCVMiniCvtColor::body(const Runtime::CallingFrame &,
...
+                                                uint32_t OutImgH,
+                                                uint32_t Interpolation) {
+  auto Src = Env.getMat(SrcMatKey);
+
+  if (!Src) {
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  cv::Mat Dst;
+  cv::resize(*Src, Dst, cv::Size(OutImgW, OutImgH), 0, 0, Interpolation);
```
Comment (dannypsnl) [inline plugins/wasmedge_opencvmini/opencvmini_func.cpp:197]: hard coded args

**Review by dannypsnl — CHANGES_REQUESTED** [review]

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp (hunk 2 above)
Comment (Charles-Schleich) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp]: Updated. I've added in the scale factors into the function definition for resize.

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp (hunk 1 above)
Comment (Charles-Schleich) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp]: Updated, Ive split normalize into two variants, 
`normalize`, expecting all arguments, including a Key to a mask,
And `normalize_no_mask`, using `cv::noArray()` for developer convenience.

An alternative is to expose a function to create an empty array in the environment, returning a key of that empty array, 
Then let the developer use that and only expose the normalize function. 

@dannypsnl 
Let me know if you prefer that, or if you are happy with the two function approach.

**Review by dannypsnl — APPROVED** [review]

**Comment by dannypsnl** [comment]
@Charles-Schleich Just notice that you didn't signed-off your commit, I recommend you use `git rebase` to ensure your commits did it, and squash into a single commit.

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp (hunk 1 above)
Comment (dannypsnl) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp]: The first one is enough, we can wrap it up in final library to get convience

**Review by hydai — CHANGES_REQUESTED** [review]
Hi @Charles-Schleich 
Please sign your commits with DCO before getting it merged. Thanks!

Ref: https://www.secondstate.io/articles/dco/

**Comment by dannypsnl** [comment]
Not just the latest commit, dco is required on all commits

**Comment by Charles-Schleich** [comment]
Hi @dannypsnl,
Thanks for the feedback, 
I've gone and retroactively signed off all my commits, 
as well as rebase-squashed my commits onto a single commit.
Let me know if this matches the intended format for contributions.

**Comment by dannypsnl** [comment]
Dco is ok now, extension test is failing, please check the details then

**Comment by Charles-Schleich** [comment]
Hi @dannypsnl 

I have fixed the failing tests, 
i see there are still failing pipelines, but i am not certain that they are related to my code.
If there is anything i need to fix, please let me know and i will get on it.

**Review by dannypsnl — APPROVED** [review]

**Comment by Charles-Schleich** [comment]
Hi @dannypsnl, 

I'm not planning on adding any more changes to this branch. 
Should i rather merge this into another branch other than master ?
Rather one of the dev / test branches ?

**Comment by dannypsnl** [comment]
Sure, @hydai I will make a `plugin/opencvmini` branch for this.

**Pull request merged**